newIDE/app/src/EventsSheet/ParameterFields/ObjectVariableField.js

```diff
@@ -112,6 +116,23 @@ export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
       [variablesContainers]
     );
 
+    const onVariableEditorApply = React.useCallback(
+      (selectedVariableName: string | null) => {
+        if (
+          selectedVariableName &&
+          selectedVariableName.startsWith(props.value)
+        ) {
+          props.onChange(selectedVariableName);
+        }
+        setEditorOpen(null);
+        // The variable editor may have refactor the events for a variable type
+        // change which may have change the currently edited instruction type.
+        if (onInstructionTypeChanged) onInstructionTypeChanged();
+        if (field.current) field.current.updateAutocompletions();
+      },
+      [onInstructionTypeChanged, props]
```

Comment:
props is a JS object that changes at every render - I'm surprised React linter is not warning you against this. This means the useCallback is useless here: it will change at every call.

Instead, destructure first and pass onChange and value.